homeassistant/components/zha/__init__.py

```diff
@@ -369,6 +410,104 @@ def zdo_command(self, tsn, command_id, args):
         pass
 
 
+class ZHAEvent(object):
+    """When you want signals instead of entities.
+    Stateless sensors such as remotes are expected to generate an event
+    instead of a sensor entity in hass.
+    """
+
+    def __init__(self, hass, cluster, discovery_info):
+        """Register callback that will be used for signals."""
+        self._hass = hass
+        self._cluster = cluster
+        self._cluster.add_listener(self)
+        self._ieee = discovery_info['endpoint'].device.ieee
+        self._ieeetail = ''.join(['%02x' % (o, ) for o in self._ieee[-4:]])
+        if discovery_info['manufacturer'] and discovery_info['model'] is not \
+                None:
+            self._id = "{}.{}_{}_{}_{}{}".format(
+                'zha',
+                slugify(discovery_info['manufacturer']),
+                slugify(discovery_info['model']),
+                self._ieeetail,
+                discovery_info['endpoint'].endpoint_id,
+                discovery_info.get('entity_suffix', '')
+            )
+        else:
+            self._id = "{}.zha_{}_{}{}".format(
+                'zha',
+                self._ieeetail,
+                discovery_info['endpoint'].endpoint_id,
+                discovery_info.get('entity_suffix', '')
+            )
+
+
+class ZHASwitchEvent(ZHAEvent):
+    """Switch / remote event for zha"""
+
+    def __init__(self, hass, cluster, discovery_info):
+        """Initialize Switch."""
+        super().__init__(hass, cluster, discovery_info)
+
+    @callback
+    def cluster_command(self, tsn, command_id, args):
+        """Handle commands received to this cluster."""
+        if command_id in (0x0000, 0x0040):
+            self._hass.bus.fire('zha.off', {'device': self._id},
+                                EventOrigin.remote)
+        elif command_id in (0x0001, 0x0041, 0x0042):
+            self._hass.bus.fire('zha.on', {'device': self._id},
+                                EventOrigin.remote)
+        elif command_id == 0x0002:
+            self._hass.bus.fire('zha.toggle', {'device': self._id},
+                                EventOrigin.remote)
+
+
+class ZHALevelEvent(ZHAEvent):
+    """Switch / remote event for zha"""
+
+    def __init__(self, hass, cluster, discovery_info):
+        """Initialize Switch."""
+        super().__init__(hass, cluster, discovery_info)
+        self._level = 0
+
+    @callback
+    def cluster_command(self, tsn, command_id, args):
+        """Handle commands received to this cluster."""
+        if command_id in (0x0000, 0x0004):  # move_to_level, -with_on_off
```

Comment:
I would love to see all things like `0x0000` and `0x0004` be extracted into constants with names like `MOVE_TO_LEVEL` etc. These constants should at least live at the top of the file, but preferably they are moved to ZHA.

Reply:
agreed. I can try to tackle that this week in separate PR's to both repos.

Reply:
Can you start with putting them in constants defined at the top of this file

Reply:
Will do

Reply:
something like?
```
cmd = LevelControl.server_commands.get(command_id, ('unknown'))[0]
if cmd in ('move_to_level', 'move_to_level_with_on_off'):
     pass
```